msal_extensions/persistence.py

```diff
@@ -115,13 +155,28 @@ def __init__(self, location, entropy=''):
 
     def save(self, content):
         # type: (str) -> None
+        data = self._dp_agent.protect(content)
         with open(self._location, 'wb+') as handle:
-            handle.write(self._dp_agent.protect(content))
+            handle.write(data)
 
     def load(self):
         # type: () -> str
-        with open(self._location, 'rb') as handle:
-            return self._dp_agent.unprotect(handle.read())
+        try:
+            with open(self._location, 'rb') as handle:
+                data = handle.read()
+            return self._dp_agent.unprotect(data)
+        except EnvironmentError as exp:  # EnvironmentError in Py 2.7 works across platform
+            if exp.errno == errno.ENOENT:
+                raise PersistenceNotFound(
+                    message=(
+                        "Persistence not initialized. "
+                        "You can recover by calling a save() first."),
+                    where=self._location,
+                    )
+            logger.exception(
```

Comment:
We know this can be one of the reasons but such a logging without explicity checking it might be misleading for cases where a different dpapi issue is causing unprotect to fail. We can either just skip this hardcoded message or if we want to log what is causing the error, we can use the a message from the caught exception instead

Reply:
Will it help if we highlight the keyword "LIKELY" at line 177? This message does not mean to be a decisive conclusion, but we are giving a most likely reason and then provide a self-troubleshooting suggestion.  #supportability